lib/src/Base/Func/UniVariateFunctionFactory.cxx

```diff
@@ -0,0 +1,82 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief This an abstract class for 1D function factories
+ *
+ *  Copyright 2005-2016 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/UniVariateFunctionFactory.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/OSS.hxx"
+#include "openturns/Exception.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+
+
+CLASSNAMEINIT(UniVariateFunctionFactory);
+
+static const Factory<UniVariateFunctionFactory> Factory_UniVariateFunctionFactory;
+
+
+/* Default constructor */
+UniVariateFunctionFactory::UniVariateFunctionFactory()
+  : PersistentObject()
+  , functionsCache_(0)
+{
+  // Nothing to do. The derived class will have to call initializeCaches().
+}
+
+/* Virtual constructor */
+UniVariateFunctionFactory * UniVariateFunctionFactory::clone() const
+{
+  return new UniVariateFunctionFactory(*this);
+}
+
+
+/* String converter */
+String UniVariateFunctionFactory::__repr__() const
+{
+  return OSS() << "class=" << getClassName();
+}
+
+
+/* The method to get the function of any order */
+UniVariateFunction UniVariateFunctionFactory::build(const UnsignedInteger order) const
+{
+  throw NotYetImplementedException(HERE) << "UniVariateFunctionFactory::build";
+}
+
+/* Cache initialization */
+void UniVariateFunctionFactory::initializeCache()
+{
+
```

Comment:
If this method has to be redefined in derived classes, a trow NotYetImplementedException(HERE) << "In UniVariateFunctionFactory::initializeCache()" is missing, otherwise at least a "// Nothing to do" is mandatory